### PR TITLE
Split out Data Movement and WebJobs pipelines

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/ci.webjobs.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger: none
```
Comment (jsquire) [inline sdk/storage/ci.webjobs.yml:3]: We should consider naming this `ci.functions.yml` to keep consistent with the other functions-specific pipelines.

Comment (seanmcc-msft) [reply sdk/storage/ci.webjobs.yml]: Fixed.

File: sdk/storage/service.projects [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+-->
+
+<Project>
+  <ItemGroup Condition="'$(SDKType)' == 'dataplane'">
+    <ProjectsToIncludeBySDKType Include="$(MSBuildThisFileDirectory)Azure.Storage.*\**\*.csproj" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(SDKType)' == 'mgmt'">
+    <ProjectsToIncludeBySDKType Include="$(MSBuildThisFileDirectory)Azure.ResourceManager.*\**\*.csproj" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(SDKType)' == 'webjobs'">
```
Comment (seanmcc-msft) [inline sdk/storage/service.projects:23]: @jsquire should I change this to 'functions' to match the ci.functions.yaml file?

File: sdk/storage/service.projects [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  SDKType values
+   - "all" will build all projects
+   - "dataplane" will build the new track 2 client/data libraries
+   - "mgmt" will build the management plane SDK.
+   - "webjobs" will build the storage WebJobs Extensions.
+-->
+
+<Project>
+  <ItemGroup Condition="'$(SDKType)' == 'dataplane'">
```
Comment (jsquire) [inline sdk/storage/service.projects:17]:
```suggestion
  <ItemGroup Condition="'$(SDKType)' == 'client'">
```

File: sdk/storage/service.projects (hunk 2 above)
Comment (jsquire) [inline sdk/storage/service.projects:23]:
```suggestion
  <ItemGroup Condition="'$(SDKType)' == 'functions'">
```

Comment (jsquire) [reply sdk/storage/service.projects:23]: Yes.   We'll also need one for `datamovement`

Comment (seanmcc-msft) [reply sdk/storage/service.projects]: Done.

File: sdk/storage/service.projects [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  Based on the SDKType variable (msbuild property or env variable) this will filter
+  the set of projects to to match what type of SDK we are interested in building.
+
+  SDKType values
+   - "all" will build all projects
+   - "dataplane" will build the new track 2 client/data libraries
+   - "mgmt" will build the management plane SDK.
+   - "webjobs" will build the storage WebJobs Extensions.
+   - "datamovement" will build the storage Data Movement packages.
```
Comment (jsquire) [inline sdk/storage/service.projects:14]:
```suggestion
   - "all" will build all projects
   - "client" will build the new track 2 client/data libraries
   - "mgmt" will build the management plane SDK.
   - "functions" will build the storage WebJobs Extensions.
   - "datamovement" will build the storage Data Movement packages.
```

**Review by jsquire — COMMENTED** [review]
Just need the same variants of the `tests.yml` file to break out the live pipelines and we should be good to do a prepare and test.

File: sdk/storage/ci.functions.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    - sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage/
+    - sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/
+    - sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/
+    - sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Queues/
+    - sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Scenario.Tests/
+
+extends:
+  template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
+  parameters:
+    ServiceDirectory: storage
```
Comment (weshaggard) [inline sdk/storage/ci.functions.yml:23]: You will need to set SDKType in each of these yml files as well. https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/ci.functions.yml#L31

Comment (seanmcc-msft) [reply sdk/storage/ci.functions.yml:24]: Done

**Comment by jsquire** [comment]
> Just need the same variants of the `tests.yml` file to break out the live pipelines and we should be good to do a prepare and test.

Still need the `tests.yml` variants.

File: sdk/storage/ci.functions.yml (hunk 5 above)
Comment (jsquire) [reply sdk/storage/ci.functions.yml:24]: Thanks, Wes.   I overlooked that!

**Review by jsquire — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
@jsquire are we good to merge even though the net - storage - functions - tests pipeline appears to have a problem?

**Comment by jsquire** [comment]
> @jsquire are we good to merge even though the net - storage - functions - tests pipeline appears to have a problem?

That's your call @seanmcc-msft.   I just wanted to kick off a pipeline to be sure that the prepare worked.   The failures that I looked at appear to be normal transient issues.   If you're comfortable, I've got no objections to merging.

**Pull request merged**